src/jdk.internal.vm.compiler/share/classes/org.graalvm.compiler.hotspot.aarch64/src/org/graalvm/compiler/hotspot/aarch64/AArch64HotSpotConstantRetrievalOp.java

```diff
@@ -116,7 +116,7 @@ public void emitCode(CompilationResultBuilder crb, AArch64MacroAssembler masm) {
         }
 
         final int before = masm.position();
-        masm.bl(before);
+        masm.bl(0);
```

Comment:
results of `test/hotspot/jtreg:tier1_compiler_aot_jvmci` before this change:
```
Test results: passed: 102; failed: 40; error: 8
```
after:
```
Test results: passed: 120; failed: 22; error: 8
```

I had quite some fun tracking this down, here are some more details: https://gist.github.com/lewurm/bdc85a15a9e735ac16aa23c1a6d0c8c7